### DISCUSSION TITLE
LESSON4: "Method eth_maxPriorityFeePerGas not supported."

### DISCUSSION THREAD

**Discussion by elmbark2023** [smartcontractkit/full-blockchain-solidity-course-py#409] [Q&A] [answered]
Good Time!

I am working on Lesson 4 and trying to connect to Ganache, but I kept getting the following error message even with the source file from git repo.
Does anybody have some clue on this?

![1637388436911](https://user-images.githubusercontent.com/1181400/142716464-809dd142-0bcb-4628-9a1b-35738f78cc39.jpg)
.

**Comment by elmbark2023** [comment, marked as answer]
OK, I figured it out, actually somebody asked it before and I believe it is a general problem related to the web3.py version.
After I installed the 5.24.0 version of web3, it worked.
but I don't understand the reason for this issue and google doesn't give me any answer either. 
hopefully author would notice this and shed some light.

**Reply by cromewar** [reply]
Hello @ziyuan I was writing the answer for you were faster than me haha, I'm glad you solved your problem. 
Bases on your error it might be due to some Cytools dependencies not fulfilled, most users had to install visual studio and it's python tools.

Anyways as your problem is solved I'll close this thread, best regards.

Note: try to use WSL it would help your a lot with this stuff ;)

**Reply by CSalle** [reply]
This one worked like a charm for me :) Thanks!

**Comment by elmbark2023** [comment]
thank you so much for your reply!

I am quite new to this field,  I was a C++ software engineer but that was 5
years ago.

btw, what is WSL? I notice web3.py package is a little bit complicated.



Vasiliy Gualoto ***@***.***> 于2021年11月20日周六 上午8:14写道：

> Hello @ziyuan <https://github.com/ziyuan> I was writing the answer for
> you were faster than me haha, I'm glad you solved your problem.
> Bases on your error it might be due to some Cytools dependencies not
> fulfilled, most users had to install visual studio and it's python tools.
>
> Anyways as your problem is solved I'll close this thread, best regards.
>
> Note: try to use WSL it would help your a lot with this stuff ;)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/smartcontractkit/full-blockchain-solidity-course-py/discussions/409#discussioncomment-1674735>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAJANWG3TCCQQQLGTHIGJ7TUM6ULPANCNFSM5INTCQEQ>
> .
>


-- 
Ziyuan Wang

**Reply by cromewar** [reply]
Yes web3 is tricky, anyways you will use frameworks like brownie or truffle which managed web3 stuff for you, almost most of them.

WSL is Windows Subsystem for linux, is a way to run a linux terminal inside your windows machine and its helps you a lot with your development stuff, please refer to the course index here in github and see "Windows support" section you'll find an article about how to setup your windows machine for smart contract development.

**Comment by elmbark2023** [comment]
I am actually developing on macbook pro, not sure if I did the correct thin
in building the environment and installing different packages,

which I found can be a headache for python.

anyway, thank you for answering my questions!!


Vasiliy Gualoto ***@***.***> 于2021年11月20日周六 上午8:53写道：

> Yes web3 is tricky, anyways you will use frameworks like brownie or
> truffle which managed web3 stuff for you, almost most of them.
>
> WSL is Windows Subsystem for linux, is a way to run a linux terminal
> inside your windows machine and its helps you a lot with your development
> stuff, please refer to the course index here in github and see "Windows
> support" section you'll find an article about how to setup your windows
> machine for smart contract development.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/smartcontractkit/full-blockchain-solidity-course-py/discussions/409#discussioncomment-1674846>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAJANWH6VKQ3K26YYWDHATDUM6Y6ZANCNFSM5INTCQEQ>
> .
>


-- 
Ziyuan Wang

**Reply by cromewar** [reply]
In that case this simplifies things a lot, I suggested wsl as Linux is Unix based, so Mac is pretty much the same on its core.. you should like to install python virtualenv and manage each project with different packages.

Best regards, Cromewar.

**Comment by elmbark2023** [comment]
Gotcha!  Thank you!

I will do some research in visual studio + virtualenv.

I am very excited in coding some web3 apps.



Vasiliy Gualoto ***@***.***> 于2021年11月20日周六 上午9:49写道：

> In that case this simplifies things a lot, I suggested wsl as Linux is
> Unix based, so Mac is pretty much the same on its core.. you should like to
> install python virtualenv and manage each project with different packages.
>
> Best regards, Cromewar.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/smartcontractkit/full-blockchain-solidity-course-py/discussions/409#discussioncomment-1674985>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAJANWBZTS5JQZAB24RI7C3UM67SNANCNFSM5INTCQEQ>
> .
>


-- 
Ziyuan Wang

**Comment by Arsenii0** [comment]
Managed to solve this problem using the suggestion from https://stackoverflow.com/questions/70051896/python-web3-and-ganache-error-method-eth-maxpriorityfeepergas-not-supported.
 
Try to rewrite `buildTransaction()` method call
```
transaction = SimpleStorage.constructor().buildTransaction(
    {
        "gasPrice": w3.eth.gas_price, # add this line
        "chainId": chain_id,
        "from": my_address,
        "nonce": nonce,
    }
)
```

**Comment by elmbark2023** [comment]
Thanks for the solution,

Now I understand this issue better. this may due to the fact that after the EIP-1559 update, it uses the priority fee for gas price, but the ganache by default still use the previous gasPrice parameter. 

As the following article suggest

https://github.com/trufflesuite/ganache/issues/939
